core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -1555,4 +1542,37 @@ namespace kagome::consensus::grandpa {
     };
     retain_if(waiting_blocks_, f);
   }
+
+  void GrandpaImpl::saveCachedVotes() {
+    CachedVotes rounds;
+    for (auto round = current_round_; round;
+         round = round->getPreviousRound()) {
+      rounds.emplace_back(CachedRound{
+          round->voterSetId(),
+          round->roundNumber(),
+          round->votes(),
+      });
+    }
+    std::ignore =
+        db_->put(storage::kGrandpaImplVotesKey, scale::encode(rounds).value());
+  }
+
+  void GrandpaImpl::findCachedVotes(VotingRound &round) {
```

Comment:
Method name does not reflect the action. Lets rename to `applyCachedVotes` or `applyCachedVotesIfAny`